Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -414,7 +414,7 @@ public bool RemoveColumn( DataGridColumn<TItem> column )
     /// Links the child row with this datagrid.
     /// </summary>
     /// <param name="row">Row to remove.</param>
-    public bool RemoveRow( DataGridRowInfo<TItem> row )
+    public bool RemoveRow( _DataGridRowInfo<TItem> row )

```

Comment:
I don't think it makes sense to mix public API and private APIs. In this case we have public methods that are working with our "private" models. 

I would leave it as previously, as `DataGridRowInfo`. And then make some fields as internal. If possible.

Reply:
This change comes form the rename of `DataGridRowInfo` to `_DataGridRowInfo`.
There is no change in functionality or accessibility or class usage

Reply:
> There is no change in functionality or accessibility or class usage

True, but it is still a breaking change. And now we have a private named class publicly exposed. 

Revert to the original name, as we don't want this kind if naming for our API.

Reply:
Ok, I reverted back and renamed created the `DataGridRowPublicInfo` ...
I still don't quite understand the conventions here and the public/private/internal logic behind this.

Why are some members marked `public`, while used only for internal stuff? Why are some of those named with `_` prefix and some are not?

Reply:
It all boils down to simplicity. I never liked too much separation of concern just for the sake of it. And your version with the new `DataGridRowPublicInfo` was going into that direction. I just find it confusing, and introduced overhead when reading the code. Making the `DataGridRowInfo` is simple enough and then hiding internal `TableRow` enough.

Please don't find this an attack onto you in any way. Both of us are still learning from each other.

Reply:
I agree this is the simpler solution.

I see you changed the `TableRow` to internal and that's exactly the place where I lack the necessary knowledge. The knowledge that cannot be extracted out of just reading the code and probably out of anywhere.

I guess it's simpler to mark everything as public, but it really makes it hard-to-impossible to see what really is supposed to be public and what can be changed without any breaking changes.

Do you have any guidelines for that? Would you consider to refactor these parts for the sake of better orientation in the code?